### PR TITLE
Improved the unit test coverage for the merge function of dynamichelper

### PR DESCRIPTION
**Pull request by xiaoyu74**
### Which issue this PR addresses:
To improve unit test coverage for dynamichelper. This ADO task can be separated to multiple PRs for each function in the dynamichelper as suggested.

### What this PR does / why we need it:
The dynamichelper is used in multiple important parts of our codebase and need better unit test coverage

### Is there any documentation that needs to be updated for this PR?
No, this PR is to meet the requirement in the ADO task [Improve unit test coverage for dynamichelper](https://msazure.visualstudio.com/AzureRedHatOpenShift/_boards/board/t/Skippy/Stories/?workitem=13525478)

### CODE REVIEW

**Review by s-amann — DISMISSED** [review]
lgtm

File: pkg/util/dynamichelper/dynamichelper_test.go [hunk 1]
```diff
@@ -8,15 +8,26 @@ import (
...
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
 )
 
 func TestMerge(t *testing.T) {
 	serviceInternalTrafficPolicy := corev1.ServiceInternalTrafficPolicyCluster
+	var terminateGracePeriod int64 = corev1.DefaultTerminationGracePeriodSeconds
+	var revisionHistoryLimit int32 = 10
+	var replicasDeployment int32 = 1
+	var progressDlineSec int32 = 600
```
Comment (darthhexx) [inline pkg/util/dynamichelper/dynamichelper_test.go:29]: Overall this lgtm but I have one small nit. We should rename this variable to `progressDeadlineSec` or something similar.

**Review by darthhexx — COMMENTED** [review]
Just one small naming convention nit.

**Review by SudoBrendan — APPROVED** [review]
LGTM

**Review by darthhexx — DISMISSED** [review]
LGTM

**Review by SudoBrendan — DISMISSED** [review]

**Review by m1kola — DISMISSED** [review]
Looks good to me. Let's merge on green CI.

**Review by s-amann — DISMISSED** [review]

**Comment by xiaoyu74** [comment]
The E2E issue has been fixed and CI is green now.

**Review by s-amann — DISMISSED** [review]

**Review by darthhexx — APPROVED** [review]
Thanks for rebasing for Go1.17. LGTM

**Pull request merged**